### PR TITLE
Nix script

### LINKED ISSUE #2385: Feature Request
**Is your feature request related to a problem? Please describe.**
Creating a installation script to simplify the installation process while setup  of the project

**Describe the solution you'd like**
Creating a nix flake to one click install the packages required for the setup of the project.

**Describe alternatives you've considered**
Creating a bash script for this but nix simplifies the installation process.

**Potential internship candidates**
Please read this if you are planning to apply for a Palisadoes Foundation internship https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by AVtheking**
**What kind of change does this PR introduce?**

Created a installation script to install the packages of the project

**Issue Number:**

Fixes #2385 

**Did you add tests for your changes?**

No

**If relevant, did you update the documentation?**

**Summary**
Created a nix flake to install the packages required to run and setup the talawa project.

**Does this PR introduce a breaking change?**
No

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-api/blob/master/CONTRIBUTING.md)?**

## Summary by CodeRabbit

- **New Features**
	- Introduced Nix flake configuration to streamline the development environment setup for the Talawa API, including necessary tools and services.
	- Configured Redis to store data in a specified directory for improved data management.

- **Chores**
	- Added a setup script to automate the installation of the Nix package manager and facilitate the environment setup across different operating systems.
	- Included multiple security vulnerability reports for various packages to enhance overall security awareness.

### CODE REVIEW

**Comment by AVtheking** [comment]
@palisadoes  @DMills27  please review.

File: flake.nix [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        devShells.default = pkgs.mkShell {
+          buildInputs = [
+            pkgs.git
+            pkgs.nodejs
+            pkgs.typescript
+            pkgs.fnm
+            pkgs.redis
+          ];
+
+          shellHook = ''
```
Comment (DMills27) [inline flake.nix:29]: Don't use a shellHook. This is what is creating the unnecessary overhead. You don't need an ephemeral shell to perform these actions; they should be done in the respective build phases.

**Comment by DMills27** [comment]
You still need to do bit a bit of work. Right off the bat I see that you used `shellHook` which is what is creating the unnecessary overhead. I implore you to look at [this example](https://zebreus.github.io/nix-website-example/) to see how you could use the the build phases to properly set this up instead. 

Additionally, your flake only installs the packages and does some minor configurations. You actually need to configure the various parts will interact with one another to get the API up and running. The end result should be a local instance of the Talawa API for whoever builds the flake and not just the various services it need to run. Here are some additional resources:

- https://gist.github.com/joepie91/b0041188c043259e6e1059d026eff301
- https://discourse.nixos.org/t/nix-build-phases-run-nix-build-phases-interactively/36090 (this is paerticularly useful for debugging build phases)
- https://tonyfinn.com/blog/nix-from-first-principles-flake-edition/nix-9-runnable-flakes/
- https://dev.to/deciduously/workstation-management-with-nix-flakes-build-a-cmake-c-package-21lp
-

**Comment by palisadoes** [comment]
@AVtheking Please talk to @xoldd about this. He has a simpler solution that he thinks should work.

**Pull request closed without merging**